src/backends/backend/python/resources/startup.py

```diff
@@ -0,0 +1,163 @@
+                        required=True,
+                        type=str,
+                        help="Triton instance name")
+    return parser.parse_args()
+
+
+lock = threading.Lock()
+cv = threading.Condition(lock)
+
+
+class PythonHost(PythonInterpreterServicer):
+    r"""
+    This class handles inference request for python script.
+    """
+
+    def __init__(self, module_path, *args, **kwargs):
+        super(PythonInterpreterServicer, self).__init__(*args, **kwargs)
+        spec = importlib.util.spec_from_file_location("triton", module_path)
+        module = importlib.util.module_from_spec(spec)
+        spec.loader.exec_module(module)
+
+        if hasattr(module, "initialize_model"):
+            self.initializer_func = module.initialize_model
+        elif hasattr(module, "triton"):
+            self.initializer_func = module.triton
+        else:
+            self.initializer_func = None
+        self.model = None
+
+    def Init(self, request, context):
+        # if self.model:
+        #     return StatusCode(code=0)
+
+        # if not self.initializer_func:
+        #     return StatusCode(code=1)
+        args = {x.key: x.value for x in request.model_command}
+        self.model = self.initializer_func(args)
+
+        # if not self.model:
+        #     return StatusCode(code=2)
+
+        # return StatusCode(code=0)
+
+    def Fini(self, request, context):
+        if hasattr(self.model, "shutdown"):
+            self.model.shutdown()
+
+        del self.model
+        with cv:
+            cv.notify()
+        # return StatusCode(code=0)
+
+    def Execute(self, request, context):
+
+        requests = request.requests
+        np_requests = []
+        for request in requests:
+            request_inputs = {}
+            for request_input in request.inputs:
+                x = request_input
+                request_inputs[x.name] = np.frombuffer(
+                    x.raw_data,
+                    dtype=protobuf_to_numpy_type(x.dtype)).reshape(x.dims)
+            np_requests.append({
+                'inputs': request_inputs,
+                'id': request.id,
+                'correlation_id': request.correlation_id,
+                'requested_output_names': request.requested_output_names
+            })
+        responses = self.model(np_requests)
+        exec_responses = []
+        for response in responses:
+            tensors = []
+            for response_name, response_value in response.items():
+                name = response_name
+                tensor = Tensor(name=name,
+                                dtype=numpy_to_protobuf_type(
+                                    response_value.dtype.type),
+                                dims=response_value.shape,
+                                raw_data=response_value.tobytes())
+                tensors.append(tensor)
+            exec_responses.append(InferenceResponse(inputs=tensors))
+        execute_response = ExecuteResponse(responses=exec_responses)
```

Comment:
Is this copying the entire exec_responses into responses? We want to construct ExecuteResponse in place with no copying.

Reply:
I don't think it is copying the responses. It is just the constructor argument.

Reply:
Are you sure? So if after line 147 you were to modify exec_responses, those modifications would also appear in execute_response?